File: javascript/1049.js
var input = require('fs').readFileSync('/dev/stdin', 'utf8');
var lines = input.split('\n');
var [a,b,c] = lines;

if(a == "vertebrado"){
  if(b == "ave"){
    if(c == "carnivoro"){
      console.log("aguia");
      
    }else if(c == "onivoro"){
        
      console.log("pomba");
    }
  }
  else if(b == "mamifero"){
    if(c == "onivoro"){
      console.log("homem");
    }else if(c == "herbivoro"){
      console.log("vaca");
    }
  }
}else if(a == "invertebrado"){
  if(b == "inseto"){
    if(c == "hematofago"){
      console.log("pulga");
    }else if(c == "herbivoro"){
      console.log("lagarta");
    }
  }else if(b == "anelideo"){
    if(c == "hematofago"){
      console.log("sanguessuga");
    }else if(c == "onivoro"){
      console.log("minhoca");
    }
  }
}

File: C++/1049.cpp
#include <iostream>

using namespace std;

int main(){
    
    string a, b, c;
    getline(cin, a);
    getline(cin, b);
    getline(cin, c);
    
    if(a[0] == 'v'){
      if(b[0] == 'a'){
        if(c[0] == 'c'){
          cout << "aguia" << endl;
        }else if(c[0] == 'o'){
          cout << "pomba" << endl;
        }
      }
      else if(b[0] == 'm'){
        if(c[0] == 'o'){
          cout << "homem" << endl;
        }else if(c[0] == 'h'){
          cout << "vaca" << endl;
        }
      }
    }else if(a[0] == 'i'){
      if(b[0] == 'i'){
        if(c[2] == 'm'){
          cout << "pulga" << endl;
        }else if(c[0] == 'h'){
          cout << "lagarta" << endl;
        }
      }else if(b[0] == 'a'){
        if(c[0] == 'h'){
          cout << "sanguessuga" << endl;
        }else if(c[0] == 'o'){
          cout << "minhoca" << endl;
        }
      }
    }
    return 0;
}